lib/reverse_proxy_plug.ex

```diff
@@ -152,42 +150,49 @@ defmodule ReverseProxyPlug do
     |> Conn.resp(status, body)
   end
 
-  @spec stream_response(Conn.t(), Keyword.t()) :: Conn.t()
-  defp stream_response(conn, opts) do
-    receive do
-      %HTTPoison.AsyncStatus{code: code} ->
-        case opts[:status_callbacks][code] do
-          nil ->
-            conn
-            |> Conn.put_status(code)
-            |> stream_response(opts)
-
-          handler ->
-            handler.(conn, opts)
-        end
-
-      %HTTPoison.AsyncHeaders{headers: headers} ->
-        headers
-        |> normalize_headers
-        |> Enum.reject(fn {header, _} -> header == "content-length" end)
-        |> Enum.concat([{"transfer-encoding", "chunked"}])
-        |> Enum.reduce(conn, fn {header, value}, conn ->
-          Conn.put_resp_header(conn, header, value)
-        end)
-        |> Conn.send_chunked(conn.status)
-        |> stream_response(opts)
-
-      %HTTPoison.AsyncChunk{chunk: chunk} ->
-        case Conn.chunk(conn, chunk) do
-          {:ok, conn} ->
-            stream_response(conn, opts)
-
-          {:error, :closed} ->
-            conn
-        end
-
-      %HTTPoison.AsyncEnd{} ->
-        conn
+  if {:module, HTTPoison} == Code.ensure_loaded(HTTPoison) do
```

Comment:
@mwhitworth This is perhaps the most controversial point in this PR. The issue here is that this behavior is inherent to HTTPoison and if I were to redesign this PR as a breaking change, this probably would be removed altogether (or moved to the HTTPoison adapter somehow).

I've chosen to not compile this section of the code if HTTPoison isn't present (it is now an optional dep). WDYT about this?

Reply:
Besides these points, I think the PR is ready for review

Reply:
I think this is fine. Even if other libraries supported this, it would need to be reworked anyway.